gnovm/tests/files/zrealm_crossrealm11.gno

```diff
@@ -0,0 +1,152 @@
+// PKGPATH: gno.land/r/crossrealm_test
+package crossrealm_test
+
+import (
+	"std"
+
+	ptests "gno.land/p/demo/tests"
+	"gno.land/p/demo/ufmt"
+	rtests "gno.land/r/demo/tests"
+	testfoo "gno.land/r/demo/tests_foo"
+)
+
+func getRealmCaller() std.Address {
+	return std.GetRealmCaller()
+}
+
+func Exec(fn func()) {
+	fn()
+}
+
+func main() {
+	// Create a map of the potential callers, this will give more understandable
+	// output than the bech32 addresses.
+	callersByAddr := make(map[std.Address]string)
+	for _, caller := range []string{
+		"user1.gno", "gno.land/r/crossrealm_test", "gno.land/r/demo/tests",
+	} {
+		addr := std.DerivePkgAddr(caller)
+		callersByAddr[addr] = caller
+	}
+
+	baseCallStack := "user1.gno -> r/crossrealm_test.main()"
+	tests := []struct {
+		callStackAdd string
+		callerFn     func() std.Address
+	}{
+		{
+			callStackAdd: "",
+			callerFn:     std.GetRealmCaller,
+		},
+		{
+			callStackAdd: " -> r/crossrealm_test.getRealmCaller()",
+			callerFn:     getRealmCaller,
+		},
+		{
+			callStackAdd: " -> p/demo/tests",
+			callerFn:     ptests.GetRealmCaller,
+		},
+		{
+			callStackAdd: " -> p/demo/tests -> p/demo/tests/subtests",
+			callerFn:     ptests.GetSubtestsCaller,
+		},
+		{
+			callStackAdd: " -> r/demo/tests",
+			callerFn:     rtests.GetRealmCaller,
+		},
+		{
+			callStackAdd: " -> r/demo/tests -> r/demo/tests/subtests",
+			callerFn:     rtests.GetSubtestsCaller,
+		},
+		{
+			callStackAdd: " -> p/demo/tests -> r/demo/tests",
+			callerFn:     ptests.GetRealmCaller,
+		},
+	}
+	for i, tt := range tests {
+		callStack := ufmt.Sprintf("%s%s -> std.GetRealmCaller()", baseCallStack, tt.callStackAdd)
+		println(ufmt.Sprintf("STACK CASE %d:\n%s", i, callStack))
+		println(ufmt.Sprintf("\tcaller is %s", callersByAddr[tt.callerFn()]))
+		Exec(func() {
+			println(ufmt.Sprintf("wrapped in r/crossrealm_test.Exec:\n\tcaller is %s", callersByAddr[tt.callerFn()]))
+		})
+		rtests.Exec(func() {
+			println(ufmt.Sprintf("wrapped in r/demo/tests.Exec:\n\tcaller is %s", callersByAddr[tt.callerFn()]))
+		})
+		ptests.Exec(func() {
+			println(ufmt.Sprintf("wrapped p/demo/tests.Exec:\n\tcaller is %s", callersByAddr[tt.callerFn()]))
+		})
+		println()
+	}
+}
+
+// Output:
```

Comment:
It is important that reviewers examine each case below carefully, to see if it matches their expectations.

Note that when `std.GetRealmCaller()` is called in a function that is passed in parameter of a realm function, the realm itself became the caller. That can be used as a trick for vault realm for instance, but this is something that should be used with care. For example let's say I pushed a contract with an `Exec` function like that :

```go
// r/naive
func Exec(fn f()) {
  fn()
  
}
```

Then any other realm could impersonate that ream and potentially steal funds. For instance : 

```go
naive.Exec(func() {
   foo20.Transfer(destAddr, 9999999999999)
})

Reply:
Let's add a test showing it, then.

Reply:
Let's include a `gno lint/vet` command to detect when an exported function accepts a function pointer as an argument. This will serve as a warning to developers, reminding them to ensure sufficient filtering asserts are in place. cc @thehowl.

Reply:
> Let's include a specific test to demonstrate it.

This PR only include `std.GetRealmCaller`, once merged, i'll make a new one for updating `foo20` to use it rather than `GetOrigCaller`.

This tests is planned to be added in this following PR, as an example of how to create a safe grc20 :)

Reply:
@moul personally I think a linter should complain to me only about things which I really need to fix, and not annoy me when I do something intentionally. So I think this "potential pitfall" is really meant to be addressed with good `gno doc` documentation on the std package, which can really be done well only after #814.

I'll create still an issue to discuss linters referring to this discussion, as I think it's good to collect ideas about potential gno-specific linters.

Reply:
Yes, I believe prioritizing #814 and creating good documentation should be our initial focus.

Additionally, adopting `// nolint` as a way to indicate intentional deviations in code can streamline our use of linters.